datafusion/core/src/physical_plan/sorts/merge.rs

```diff
@@ -227,11 +239,21 @@ impl<C: Cursor> SortPreservingMergeStream<C> {
             if self.advance(stream_idx) {
                 self.loser_tree_adjusted = false;
                 self.in_progress.push_row(stream_idx);
-                if self.in_progress.len() < self.batch_size {
+
```

Comment:
This is the hot path for SortPreservingMerge -- I suspect adding an extra check for fetch won't impact performance in any measurable way, but it might be worth while checking the benchmarks

Reply:
Makes sense, otherwise we could maybe need two separate implementations (with/without fetch)

Reply:
As benchmarks showed no significant difference, I kept this check in the hot path.